### ISSUE TITLE
Spell Checking is not working on latest version of VSCode

### ISSUE THREAD

**Issue by r-kojima** [streetsidesoftware/vscode-spell-checker#3820]
Type: Bug

Title: Spell Checking is not working on latest version of VS Code

Description
The VS Code Spell Checker extension stopped working after updating to the latest version of VS Code.

After upgrading Code Spell Checker to `v4.0.20` and restarting VS Code, spell checking suddenly stopped working.
When I downgraded Code Spell Checker to `v3.0.1` and restarted VS Code, spell checking started working properly again.

Environment: 
Extension version: 4.0.20
VS Code version: Code 1.95.3 (Universal) (f1a4fb101478ce6ec82fe9627c43efbf9e98c813, 2024-11-13T14:50:04.152Z)
OS version: Darwin arm64 23.3.0

**Comment by Jason3S** [comment]
@r-kojima,

I need a bit more information.

What do you mean by not working?
What type of file do you have open?

A screen shot would be useful.

**Comment by r-kojima** [comment]
@Jason3S 
I apologize for the insufficient information.
When opening TypeScript files, the markers for misspelled words are no longer being displayed.

Here's how it looks in v4.0.20:
![スクリーンショット 2024-11-20 18 35 46](https://github.com/user-attachments/assets/4ec73088-2cee-4a44-addc-d8158cabfb65)


In v3.0.1, spell checking works as shown below:
![スクリーンショット 2024-11-20 18 34 41](https://github.com/user-attachments/assets/0bf70627-35f8-4482-95f3-e04f41ea942c)

Additionally, I just noticed that when I upgraded Code Spell Checker to version 4.0.20, the following error appeared in VS Code. Please check:
![スクリーンショット 2024-11-20 18 35 00](https://github.com/user-attachments/assets/f6395aff-a9b1-4cb5-b151-935719917561)

**Comment by Jason3S** [comment]
@r-kojima,

Thank you.

 
> Here's how it looks in v4.0.20: 
> ![スクリーンショット 2024-11-20 18 35 46](https://private-user-images.githubusercontent.com/55544166/388036359-4ec73088-2cee-4a44-addc-d8158cabfb65.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwOTgxMDEsIm5iZiI6MTczMjA5NzgwMSwicGF0aCI6Ii81NTU0NDE2Ni8zODgwMzYzNTktNGVjNzMwODgtMmNlZS00YTQ0LWFkZGMtZDgxNThjYWJmYjY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIwVDEwMTY0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThmZTkxZTNmMWUwMTRkYzhlOWVhOGQxZWM4NTk4ODBhNGFlNGU0Nzc1YjU2NDMyNmNmMTkzZmYxOWZhYmRkY2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.l0k02CeIeCDs_XW2bHgPFAdqOW-vNne45a6V7siEfMQ)

When v4.0.20 is installed, please click `{}` next to `TypeScript`. You should get a popup like this:
<img width="315" alt="image" src="https://github.com/user-attachments/assets/03157b31-0f4f-424d-b39b-f998df2dde6c">

Click the `menu` item next to `Spell`.

You should get a menu like this:
<img width="597" alt="image" src="https://github.com/user-attachments/assets/26bf47ee-5baf-4512-a744-c42eb5680c90">

Could you make a screen shot of that menu?

Please pick `Show File Info`:
<img width="165" alt="image" src="https://github.com/user-attachments/assets/b56b3f3a-01d0-421f-97df-0ab708710737">

Something like this should show up:
<img width="362" alt="image" src="https://github.com/user-attachments/assets/1c91f76f-d6b2-4035-85d6-0f7ee9a39cec">


 
> Additionally, I just noticed that when I upgraded Code Spell Checker to version 4.0.20, the following error appeared in VS Code. Please check: 
> ![スクリーンショット 2024-11-20 18 35 00](https://private-user-images.githubusercontent.com/55544166/388036411-f6395aff-a9b1-4cb5-b151-935719917561.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwOTgxMDEsIm5iZiI6MTczMjA5NzgwMSwicGF0aCI6Ii81NTU0NDE2Ni8zODgwMzY0MTEtZjYzOTVhZmYtYTliMS00Y2I1LWIxNTEtOTM1NzE5OTE3NTYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIwVDEwMTY0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEzOGVjNjgyOWMyMWJhMTg3ZGUwMmZjYzNkZjdiMGYzMzY3OTc2NTQ2ZjczNWNjMzQwNmJjMzI4NjIwYjdhYjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ioDFm-aRVV_4gED0fVgGIe7D3RAN5-GclInN0aDEL2A)

This is a known VS Code bug: [Error: `No view is registered with id` might be caused by a race condition · Issue #225945 · microsoft/vscode](https://github.com/microsoft/vscode/issues/225945#issuecomment-2359627001)

**Comment by r-kojima** [comment]
@Jason3S 
Thank you for the reply.
I've taken screenshots, so please review them.
![スクリーンショット 2024-11-21 10 06 36](https://github.com/user-attachments/assets/7f6fd092-8e26-4217-a612-52e6519c740b)
![スクリーンショット 2024-11-21 10 06 46](https://github.com/user-attachments/assets/1b0f27c2-70c1-428b-9333-eac4fc0f2e46)


Also, thank you for telling me about the VS Code bug. The popup disappeared after restarting VS Code.

**Comment by Jason3S** [comment]
@r-kojima,

The file is blocked by the configuration. The issue is most likely in `.cspell/cspell.json` with the `files` setting.

![image](https://github.com/user-attachments/assets/50eb674d-da1b-4284-8988-3590e53e4764)

**Comment by r-kojima** [comment]
@Jason3S 
Removing the "files" line in cspell.json resolved the issue. Thank you.
By the way, I was originally specifying it like this. What's wrong w
![スクリーンショット 2024-11-21 17 09 09](https://github.com/user-attachments/assets/ca90cfe0-f5e7-40f6-8a13-af455db60d43)
ith this configuration?

It didn't work for spell checking on any files, not just the sample `src/tsst.ts` I created.

**Comment by Jason3S** [comment]
@r-kojima,

Globs -- except for the ones starting with `**/` -- are relative to the location of the `cspell.json` file. Since the config file was in `.cspell`, files would have only matched against `.cspell/src/**/*.{ts,tsx,js,jsx,json}` files.

To fix this, set `"globRoot": ".."`.

FYI: `./` is not part of the glob standard. The spell checker tries it best to get it to work. `"src/**/*.{ts,tsx,js,jsx,json}"` would be better.
